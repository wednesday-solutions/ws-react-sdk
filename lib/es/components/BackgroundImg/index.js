import{_ as r,a as e}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"../_rollup-plugin-inject-process-env-1259e43d.js";import o from"react";import t from"prop-types";import n from"styled-components";var i,a=n.div(i||(i=r(["\n  width: ",";\n  max-height: ",";\n  background-image: url(",");\n  background-position: 0 0;\n  background-repeat: no-repeat;\n  background-size: cover;\n"])),(r=>r.width||"60%"),(r=>r.maxHeight||"100%"),(r=>r.img));function p(r){return o.createElement(a,e({"data-testid":"background-img"},r))}p.propTypes={img:t.string};export default p;
