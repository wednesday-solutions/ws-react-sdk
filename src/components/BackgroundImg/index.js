/**
 *
 * BackgroundImg
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ImageContainer = styled.div`
  width: ${props => props.width || `60%`};
  max-height: ${props => props.maxHeight || `100%`};
  background-image: url(${props => props.img});
  background-position: 0 0;
  background-repeat: no-repeat;
  background-size: cover;
`;

function BackgroundImg(props) {
  return <ImageContainer data-testid={'background-img'} {...props} />;
}

BackgroundImg.propTypes = {
  img: PropTypes.string
};

export default BackgroundImg;
