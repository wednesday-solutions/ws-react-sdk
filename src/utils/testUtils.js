import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
