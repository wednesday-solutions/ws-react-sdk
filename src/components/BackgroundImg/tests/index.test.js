/**
 *
 * Tests for BackgroundImg
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import BackgroundImg from '../index';

describe('<BackgroundImg />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<BackgroundImg />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 BackgroundImg component', () => {
    const { getAllByTestId } = renderWithIntl(<BackgroundImg />);
    expect(getAllByTestId('background-img').length).toBe(1);
  });
});
