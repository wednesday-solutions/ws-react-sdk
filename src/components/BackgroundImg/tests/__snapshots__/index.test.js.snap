// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundImg /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ euZnZe"
      data-testid="background-img"
    />
  </div>
</body>
`;
